python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -1155,12 +1165,13 @@ def nested_resources(
     @cached_method
     def get_resource_definition(self) -> ConfigurableResourceFactoryResourceDefinition:
         return ConfigurableResourceFactoryResourceDefinition(
-            self.__class__,
+            self.resource_cls,
```

Comment:
cc @benpankow  I had to change this to `self.resource_cls` so that the class type of the resource thats turned into the PartialResource would be passed through, instead of the `PartialResource` type. I did a quick look through the other places `PartialResource` is used and didn't see anything that would make this change breaking, but want to confirm with you that this is an ok change to make

Reply:
This seems ok to me 👍